usaspending_api/agency/v2/views/federal_account_list.py

```diff
@@ -44,37 +52,64 @@ def format_results(self, rows):
             ]
-            item["obligated_amount"] = sum([x["obligated_amount"] for x in item["children"]])
-            item["gross_outlay_amount"] = sum([x["gross_outlay_amount"] for x in item["children"]])
+            item["obligated_amount"] = sum(x["obligated_amount"] for x in item["children"])
+            item["gross_outlay_amount"] = sum(x["gross_outlay_amount"] for x in item["children"])
+            item["total_budgetary_resources"] = sum(x["total_budgetary_resources"] for x in item["children"])
             item["children"] = sorted(item["children"], key=lambda x: x[self.pagination.sort_key], reverse=order)
         accounts = sorted(accounts, key=lambda x: x[self.pagination.sort_key], reverse=order)
         return accounts
 
     @cache_response()
     def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self.submission_ids = get_latest_submission_ids_for_fiscal_year(self.fiscal_year)
         self.sortable_columns = ["name", "obligated_amount", "gross_outlay_amount"]
         self.default_sort_column = "obligated_amount"
-        results = self.format_results(self.get_federal_account_list())
-        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = self.format_results(self.get_tbr_from_file_a_queryset(), self.get_federal_account_list_from_file_b())
+        results_length = len(results)
+        page_metadata = get_pagination_metadata(results_length, results_length, self.pagination.page)
+
         return Response(
             {
                 "toptier_code": self.toptier_code,
                 "fiscal_year": self.fiscal_year,
                 "page_metadata": page_metadata,
-                "results": results[self.pagination.lower_limit : self.pagination.upper_limit],
+                "results": results,
                 "messages": self.standard_response_messages,
             }
         )
 
-    def get_federal_account_list(self) -> List[dict]:
-        submission_ids = get_latest_submission_ids_for_fiscal_year(self.fiscal_year)
+    def get_tbr_from_file_a_queryset(self):
+        """
+        Query Total Budgetary Resources from File A
+        """
+        filters, bureau_info_subquery = self.get_common_query_objects("treasury_account_identifier")
+        filters.append(
+            Q(submission_id__in=self.submission_ids)
+        )
+
+        results = (
+            (AppropriationAccountBalances.objects.filter(*filters))
+            .values(
+                "total_budgetary_resources_amount_cpe",
+                "treasury_account_identifier__tas_rendering_label",
+            )
+            .annotate(
+                bureau_info=bureau_info_subquery
+            )
+        )
+        return results
+
+    def get_federal_account_list_from_file_b(self) -> List[dict]:
+        _, bureau_info_subquery = self.get_common_query_objects("treasury_account")
```

Comment:
Any reason you have the `Q(treasury_account__funding_toptier_agency=self.toptier_agency),` filter defined in this method, but then in the TBR method you are using `Q(**{f"{treasury_account_keyword}__federal_account__parent_toptier_agency": self.toptier_agency}),`?

Reply:
@sethstoudenmier Maybe this can be disregarded, or reframed based on the below:

Each queryset function (FA list from B, TBR from A) is calling `get_common_query_objects` and it's the `get_common_query_objects` function that uses
`Q(**{f"{treasury_account_keyword}__federal_account__parent_toptier_agency": self.toptier_agency})`. 


With a different field name for the`TreasuryAppropriationAccount` model FK in each of the File A and File B model objects. 

`get_federal_account_list_from_file_b` uses `get_common_query_objects` I guess to get the bureau info subqery. But then also does its own filtering by agency via `FinancialAccountsByProgramActivityObjectClass.treasury_account__funding_toptier_agency=self.toptier_agency`

So maybe it begs the questions: _What is the **right** way to filter File A and/or File B objects by Agency? Why is it being done differently in different places here (e.g. via the TAS's funding toptier agency, or via the TAS'S FA'S parent toptier agency)? And do we need to be consistent in how this filtering is applied to File A or File B records?_